crucible-mir/src/Mir/JSON.hs

```diff
@@ -232,10 +230,17 @@ instance FromJSON Collection where
 instance FromJSON Fn where
     parseJSON = withObject "Fn" $ \v -> do
       args <- v .: "args"
+      abi <- v .: "abi" >>= \abi -> case abi of
+        RustCall _ -> do
+          spreadArg <- v .: "spread_arg"
+          case spreadArg of
+            Just i -> return $ RustCall (RcSpreadArg i)
+            Nothing -> return $ RustCall RcNoSpreadArg
+        _ -> return abi
```

Comment:
There's a lot of infix operators on the `abi <- v .: "abi" >>= \abi -> case abi of` line. How about splitting this up a bit:

```hs
      abi <- v .: "abi"
      let abi' =
            case abi of
              RustCall _ -> do
                spreadArg <- v .: "spread_arg"
                case spreadArg of
                  Just i -> return $ RustCall (RcSpreadArg i)
                  Nothing -> return $ RustCall RcNoSpreadArg
              _ -> return abi
```

Reply:
Fixed in 579afd3857198af9fc108870f387483beb58c677